lib/scope.js

```diff
@@ -88,12 +90,50 @@ ScopeDefinition.prototype.related = function(receiver, scopeParams, condOrRefres
     // It either doesn't hit the cache or refresh is required
     var params = mergeQuery(actualCond, scopeParams, {nestedInclude: true});
     var targetModel = this.targetModel(receiver);
+
+    // If there is a through model
+    // run another query to apply filter on relatedModel(targetModel)
+    // see github.com/strongloop/loopback-datasource-juggler/issues/166
+    var scopeOnRelatedModel = params.collect &&
+        params.include.scope !== null &&
+        typeof params.include.scope === 'object';
+    if (scopeOnRelatedModel) {
+      var filter = params.include;
+      // The filter applied on relatedModel
+      var queryRelated = filter.scope;
+      delete params.include.scope;
+    };
+
     targetModel.find(params, options, function(err, data) {
       if (!err && saveOnCache) {
         defineCachedRelations(self);
         self.__cachedRelations[name] = data;
       }
-      cb(err, data);
+
+      if (scopeOnRelatedModel === true) {
+        var relatedModel = targetModel.relations[filter.relation].modelTo;
+        var IdKey = idName(relatedModel);
+
+        // Merge queryRelated filter and targetId filter
+        var buildWhere = function() {
+          var IdKeyCondition = {};
+          IdKeyCondition[IdKey] = collectTargetIds(data, IdKey);
+          var mergedWhere = {
+            and: [IdKeyCondition, queryRelated.where],
+          };
+          return mergedWhere;
+        };
+        if (queryRelated.where !== undefined) {
+          queryRelated.where = buildWhere();
+        } else {
+          queryRelated.where = {};
+          queryRelated.where[IdKey] = collectTargetIds(data, IdKey);
+        }
+
+        relatedModel.find(queryRelated, cb);
```

Comment:
:warning: This `.find` does not include the `options` parameter (eventual transactions) :warning:

Reply:
@cadesalaberry thanks for catching it, this PR is a backport, there is another PR fixes the `options`, I will backport that one as well when have time.

Reply:
Could you link the PR ?

It appears that https://github.com/strongloop/loopback-datasource-juggler/pull/1496 also solves the issue.

Reply:
@cadesalaberry https://github.com/strongloop/loopback-datasource-juggler/pull/1426
Thanks https://github.com/strongloop/loopback-datasource-juggler/pull/1496/files is good, while I would like to backport that PR since it's consistent with version 3.x.

Reply:
@jannyHou Hey there, do you have any ETA on the backport?